File: solving/03-힙/더 맵게-힙/06-jiwoung.java
import java.util.*;
class Solution {
      public int solution(int[] scoville, int K) {
        int answer = 0;

        int first=0;
        int second=0;
        PriorityQueue minq = new PriorityQueue();
        for (int i=0;i<scoville.length;i++){
            minq.offer(scoville[i]);
        }

while ((Integer)minq.peek()<K){
    if (scoville.length-1==answer){
        answer=-1;
        break;
    }

    first= (Integer) minq.poll();
    second= (Integer) minq.poll();
    minq.offer(first+(second*2));
    answer++;
}
        return answer;
    }


}

File: solving/디스크컨트롤러-힙/06-jiwoung.kt
import java.util.*
class Solution {
    fun solution(jobs: Array<IntArray>): Int {
        var answer = 0
        var time = 0
        var waittingQueue = PriorityQueue<Job>(compareBy({ it.start }))

        for (i in 0..jobs.size - 1) {
            var job = Job(jobs[i][0], jobs[i][1])
            waittingQueue.offer(job)
        }
        while (!waittingQueue.isEmpty()) {
            var readyQueue = PriorityQueue<Job>(compareBy { it.during })
            while (!waittingQueue.isEmpty() && time >= waittingQueue.peek().start) {
                readyQueue.offer(waittingQueue.poll())
            }
            if (readyQueue.isEmpty()) {
                time++
                continue
            }
            time += readyQueue.peek().during
            answer += time - readyQueue.peek().start
            readyQueue.poll()

            for (i in 0..readyQueue.size - 1) {
                waittingQueue.offer(readyQueue.poll())
            }
        }
        answer /= jobs.size
        return answer
    }
}

data class
Job(
    var start: Int = 0,
    var during: Int = 0
)
